jfix-stdlib-concurrency/src/test/java/ru/fix/stdlib/concurrency/futures/PendingFutureLimiterTest.java

```diff
@@ -0,0 +1,344 @@
+package ru.fix.stdlib.concurrency.futures;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Field;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.awaitility.Awaitility.await;
+import static org.awaitility.Awaitility.with;
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * @author Kamil Asfandiyarov
+ */
+public class PendingFutureLimiterTest {
+
+    private static final Logger log = LoggerFactory.getLogger(PendingFutureLimiterTest.class);
+    private static final long FUTURE_LIMITER_TIMEOUT_MINUTES = 15;
+
+    private final Executor executor = Executors.newFixedThreadPool(20);
+    private final AtomicInteger globalCounter = new AtomicInteger();
+    private final AtomicBoolean startTasksFlag = new AtomicBoolean();
+
+    private CompletableFuture<Void> createTask() {
+        return CompletableFuture.runAsync(() -> {
+            try {
+                synchronized (startTasksFlag) {
+                    while (!startTasksFlag.get()) {
+                        startTasksFlag.wait();
+                    }
+                }
+                globalCounter.incrementAndGet();
+            } catch (Exception exc) {
+                log.error(exc.getMessage(), exc);
+            }
+        }, executor);
+    }
+
+    @BeforeEach
+    public void clean() {
+        globalCounter.set(0);
+        startTasksFlag.set(false);
+    }
+
+    @Test
+    public void block_on_task_when_pending_count_bigger_than_max_border() throws Exception {
+
+        PendingFutureLimiter limiter = new LimiterBuilder()
+                .enqueueTasks(3)
```

Comment:
it is unclear to the reader that maxPendingCount is 3. Suggestion: in test make explicit configuration 
```
PendingFutureLimiter limiter = new LimiterBuilder()
                .maxPendingCount(3)
                .enqueueTasks(3)
```

Reply:
Added